File: dotnet/logging-standards/src/OrderServiceBad.cs
using System;
using System.Collections.Generic;
using Microsoft.Extensions.Logging;

namespace FitnessFunctions.Logging.Bad;

/// <summary>
/// OrderServiceBad — INTENTIONALLY NON-COMPLIANT
///
/// This class violates every logging standard enforced by the fitness function.
/// Run Semgrep against this file to see all violations detected:
///
///   semgrep --config .semgrep/logging-rules.yml src/OrderServiceBad.cs
///
/// Expected violations:
///   Line 25 — no-console-writeline           (Console.WriteLine)
///   Line 34 — no-string-interpolation-in-log ($"..." in LogInformation)
///   Line 41 — no-string-interpolation-in-log ($"..." in LogWarning)
///   Line 50 — no-string-concat-in-log        (string + in LogError)
///   Line 52 — no-console-writeline           (Console.WriteLine in catch)
/// </summary>
public class OrderServiceBad
{
    // ILogger is injected correctly here — the violations are in the method bodies
    private readonly ILogger<OrderServiceBad> _logger;
    private readonly Dictionary<string, Order> _orders = new();

    public OrderServiceBad(ILogger<OrderServiceBad> logger)
    {
        _logger = logger;
        // VIOLATION: Console.WriteLine at construction time
        Console.WriteLine("OrderServiceBad created");
    }

    public Order ProcessOrder(string orderId, string userId)
    {
        // VIOLATION: string interpolation in log call — eagerly evaluated free text
        _logger.LogInformation($"Processing order {orderId} for user {userId}");

        if (!_orders.TryGetValue(orderId, out var order))
        {
            // VIOLATION: string interpolation in LogWarning
            _logger.LogWarning($"Order {orderId} not found for user {userId}");
            throw new KeyNotFoundException($"Order {orderId} not found");
        }

        try
        {
            ChargePayment(order);
            return order;
        }
        catch (InvalidOperationException ex)
        {
            // VIOLATION: string concatenation in LogError
            _logger.LogError("Payment failed for order " + orderId + ": " + ex.Message);
            // VIOLATION: Console.WriteLine as well
            Console.WriteLine("ERROR: " + ex.Message);
            throw; // log-and-rethrow: duplicate noise in log aggregation
        }
    }

    private static void ChargePayment(Order order)
    {
        if (order.Amount > 10_000)
            throw new InvalidOperationException("Amount exceeds limit");

        order.Status = "charged";
    }
}

public record Order(string OrderId, string UserId, decimal Amount)
{
    public string Status { get; set; } = "pending";
}

File: Node/logging-standards/src/order-service-bad.js
'use strict';
/**
 * src/order-service-bad.js — INTENTIONALLY NON-COMPLIANT
 *
 * Violates every logging standard enforced by the fitness function.
 * Run Semgrep to see all violations:
 *
 *   semgrep --config .semgrep/logging-rules.yml src/order-service-bad.js
 *
 * Expected violations:
 *   Line 20 — no-console-log            (console.log)
 *   Line 26 — no-raw-pino-require       (require('pino') directly)
 *   Line 33 — no-template-literal-in-log (template literal in log.info)
 *   Line 40 — no-console-warn           (console.warn)
 *   Line 49 — no-console-error          (console.error)
 *   Line 52 — no-template-literal-in-log (template literal in log.error)
 */

const orders = new Map();

// VIOLATION: console.log instead of structured logger
console.log('order-service-bad module loaded');

// VIOLATION: require pino directly instead of the shared factory
const pino = require('pino');
const log = pino();

function processOrder(orderId, userId) {
  // VIOLATION: template literal in log call — free text, not queryable fields
  log.info(`Processing order ${orderId} for user ${userId}`);

  const order = orders.get(orderId);
  if (!order) {
    // VIOLATION: console.warn instead of structured logger
    console.warn('Order not found:', orderId);
    return { status: 'not_found' };
  }

  try {
    const result = chargePayment(order);
    log.info(`Order ${orderId} charged successfully`);
    return result;
  } catch (err) {
    // VIOLATION: console.error instead of structured logger
    console.error('Payment failed for order', orderId, err.message);
    // VIOLATION: template literal in log.error — and this is also log-and-rethrow
    log.error(`Payment failed: ${err.message}`);
    throw err;
  }
}

function chargePayment(order) {
  if (order.amount > 10_000) {
    throw new Error('Amount exceeds limit');
  }
  return { status: 'charged', amount: order.amount };
}

module.exports = { processOrder };
